src/analysis/processing/qgsrasteranalysisutils.cpp

```diff
@@ -0,0 +1,175 @@
+/***************************************************************************
+  qgsrasteranalysisutils.cpp
+  ---------------------
+  Date                 : June 2018
+  Copyright            : (C) 2018 by Nyall Dawson
```

Comment:
@nyalldawson  , I deliberately refereed to you here as I don't feel right to attribute your vast code improvements to myself :smile: